stingray/crossspectrum.py

```diff
@@ -0,0 +1,378 @@
+        lc1: lightcurve.Lightcurve object, optional, default None
+            The first light curve data for the channel/band of interest.
+
+        lc2: lightcurve.Lightcurve object, optional, default None
+            The light curve data for the reference band.
+
+        norm: {'frac', 'abs', 'leahy', 'none'}, default 'none'
+            The normalization of the (real part of the) cross spectrum.
+
+        Attributes
+        ----------
+        freq: numpy.ndarray
+            The array of mid-bin frequencies that the Fourier transform samples
+
+        cs: numpy.ndarray
+            The array of cross spectra (complex numbers)
+
+        df: float
+            The frequency resolution
+
+        m: int
+            The number of averaged cross-spectra amplitudes in each bin.
+
+        n: int
+            The number of data points/time bins in one segment of the light
+            curves.
+
+        nphots1: float
+            The total number of photons in light curve 1
+
+        nphots2: float
+            The total number of photons in light curve 2
+        """
+
+        assert isinstance(norm, str), "norm is not a string!"
+
+        assert norm.lower() in ["frac", "abs", "leahy", "none"], \
+                "norm must be 'frac', 'abs', 'leahy', or 'none'!"
+
+        self.norm = norm.lower()
+
+        ## check if input data is a Lightcurve object, if not make one or
+        ## make an empty Crossspectrum object if lc1 == None or lc2 == None
+        if lc1 is None or lc2 is None:
+            if lc1 is not None or lc2 is not None:
+                 print("You can't do a cross spectrum with just one "
+                         "light curve!")
+            # else:
+            #      print("Please specify input light curves!")
+            self.freq = None
+            self.cs = None
+            self.df = None
+            self.nphots1 = None
+            self.nphots2 = None
+            self.m = 1
+            self.n = None
+            return
+        self._make_crossspectrum(lc1, lc2)
+
+    def _make_crossspectrum(self, lc1, lc2):
+
+        ## make sure the inputs work!
+        assert isinstance(lc1, lightcurve.Lightcurve), \
+                        "lc1 must be a lightcurve.Lightcurve object!"
+        assert isinstance(lc2, lightcurve.Lightcurve), \
+                        "lc2 must be a lightcurve.Lightcurve object!"
+
+
+        ## total number of photons is the sum of the
+        ## counts in the light curve
+        self.nphots1 = np.sum(lc1.counts)
+        self.nphots2 = np.sum(lc2.counts)
+
+
+        ## the number of data points in the light curve
+        assert lc1.counts.shape[0] == lc2.counts.shape[0], \
+            "Light curves do not have same number of time bins per segment."
+        assert lc1.dt == lc2.dt, \
+                "Light curves do not have same time binning dt."
+        self.n = lc1.counts.shape[0]
+
+        assert lc1.tseg == lc2.tseg, "Light curves do not have same tseg."
```

Comment:
If the `dt` and number of bins of both light curves are fixed, is it possible that `tseg` can be different? I can not think of an example while writing tests.